packages/iTwinUI-react/src/core/Tree/Tree.tsx

```diff
@@ -224,40 +244,141 @@ export const Tree = <T,>(props: TreeProps<T>) => {
     return [flatList, firstLevelNodes];
   }, [data, getNode]);
 
-  return (
-    <ul
-      className={cx('iui-tree', className)}
-      role='tree'
-      onKeyDown={handleKeyDown}
-      ref={treeRef}
-      tabIndex={0}
-      onFocus={() => {
-        const items = getFocusableNodes();
-        if (items.length > 0) {
-          items[focusedIndex.current]?.focus();
-        }
-      }}
-      {...rest}
-    >
-      {flatNodesList.map((flatNode) => (
+  const itemRenderer = React.useCallback(
+    (index: number) => {
+      const node = flatNodesList[index];
+      return (
         <TreeContext.Provider
-          key={flatNode.nodeProps.nodeId}
+          key={node.nodeProps.nodeId}
           value={{
-            nodeDepth: flatNode.depth,
-            subNodeIds: flatNode.subNodeIds,
+            nodeDepth: node.depth,
+            subNodeIds: node.subNodeIds,
             groupSize:
-              flatNode.depth === 0
+              node.depth === 0
                 ? firstLevelNodesList.length
-                : flatNode.parentNode?.subNodeIds?.length ?? 0,
-            indexInGroup: flatNode.indexInGroup,
-            parentNodeId: flatNode.parentNode?.nodeProps.nodeId,
+                : node.parentNode?.subNodeIds?.length ?? 0,
+            indexInGroup: node.indexInGroup,
+            parentNodeId: node.parentNode?.nodeProps.nodeId,
+            scrollToParent: node.parentNode
+              ? () => {
+                  const parentNodeId = node.parentNode?.nodeProps.nodeId;
```

Comment:
you check if `node.parentNode` is defined above, no need here `?`

Reply:
Tell that to lint 
![image](https://user-images.githubusercontent.com/36186912/179551453-957e3ea7-8530-40d2-8af0-e3c0bcb63859.png)